cmake/kokkos_arch.cmake

```diff
@@ -588,16 +588,18 @@ IF (KOKKOS_ENABLE_SYCL)
 ENDIF()
 
 # Check support for device_global variables
-# FIXME_SYCL Even if SYCL_EXT_ONEAPI_DEVICE_GLOBAL is defined, we still can't
-#            use device global variables with shared libraries
-IF(KOKKOS_ENABLE_SYCL AND NOT BUILD_SHARED_LIBS)
+# FIXME_SYCL If SYCL_EXT_ONEAPI_DEVICE_GLOBAL is defined, we can
+#            use device global variables with shared libraries using the
+#            non-separable compilation implementation. We get compile-time
+#            errors for the separable compilation implementation in either case.
```

Comment:
```suggestion
# FIXME_SYCL If SYCL_EXT_ONEAPI_DEVICE_GLOBAL is not defined and we
#            are building shared libraries
#            use device global variables with shared libraries using the
#            non-separable compilation implementation. We get compile-time
#            errors for the separable compilation implementation in either case.
```

Reply:
This suggestion might be (mis)interpreted as "When  `SYCL_EXT_ONEAPI_DEVICE_GLOBAL` is defined, shared libraries can be used." I'd prefer to be more specific and emphasize that we still need the non-separable implementation for shared libraries to be supported (which we need for functionalities in `KokkosKernels`) and this motivates the choice below.

Reply:
Feel free to rephrase.  I just don't understand the comment as it is now.